tools/fuzzer.cpp

```diff
@@ -0,0 +1,211 @@
+////////////////////////////////////////////////////////////////////////////////
+/// @brief Library to build up VPack documents.
+///
+/// DISCLAIMER
+///
+/// Copyright 2022 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Julia Puget
+/// @author Copyright 2022, ArangoDB GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+#include <iostream>
+#include <string>
+
+#include "velocypack/vpack.h"
+#include "velocypack/velocypack-exception-macros.h"
+#include <random>
+#include <thread>
+
+using namespace arangodb::velocypack;
+
+enum class Format {
+  VPACK, JSON
+};
+
+enum RandomBuilderAdditions {ADD_ARRAY=0, ADD_OBJECT, ADD_BOOLEAN, ADD_STRING, ADD_NULL, ADD_UINT64, ADD_INT64, ADD_DOUBLE};
+
+struct RandomGenerator {
+  RandomGenerator() : mt64{rd()} {}
+
+  std::random_device rd;
+  std::mt19937_64 mt64;
+};
+
+static void usage(char* argv[]) {
+  std::cout << "Usage: " << argv[0] << " [OPTIONS] [ITERATIONS]"
```

Comment:
<threads> are not mentioned here, so it is unclear how to specify the number of threads just from the `usage` output.